Civi/Api4/Service/Spec/Provider/ActivitySpecProvider.php

```diff
@@ -23,39 +23,45 @@ class ActivitySpecProvider implements Generic\SpecProviderInterface {
   public function modifySpec(RequestSpec $spec) {
     $action = $spec->getAction();
 
-    $field = new FieldSpec('source_contact_id', 'Activity', 'Integer');
-    $field->setTitle(ts('Source Contact'));
-    $field->setLabel(ts('Added by'));
-    $field->setDescription(ts('Contact who created this activity.'));
-    $field->setRequired($action === 'create');
-    $field->setFkEntity('Contact');
-    $field->setInputType('EntityRef');
-    $spec->addFieldSpec($field);
-
-    $field = new FieldSpec('target_contact_id', 'Activity', 'Array');
-    $field->setTitle(ts('Target Contacts'));
-    $field->setLabel(ts('With Contact(s)'));
-    $field->setDescription(ts('Contact(s) involved in this activity.'));
-    $field->setFkEntity('Contact');
-    $field->setInputType('EntityRef');
-    $field->setInputAttrs(['multiple' => TRUE]);
-    $spec->addFieldSpec($field);
-
-    $field = new FieldSpec('assignee_contact_id', 'Activity', 'Array');
-    $field->setTitle(ts('Assignee Contacts'));
-    $field->setLabel(ts('Assigned to'));
-    $field->setDescription(ts('Contact(s) assigned to this activity.'));
-    $field->setFkEntity('Contact');
-    $field->setInputType('EntityRef');
-    $field->setInputAttrs(['multiple' => TRUE]);
-    $spec->addFieldSpec($field);
+    // The database default '1' is problematic as the option list is user-configurable,
+    // so activity type '1' doesn't necessarily exist. Best make the field required.
+    $spec->getFieldByName('activity_type_id')
+      ->setDefaultValue(NULL)
+      ->setRequired(TRUE);
```

Comment:
I think `status` would need the same treatment? I can still blank it out in a search display.

Reply:
Yea, that's what I meant by having a way for the api to understand `not null` separate from `required`. IMO we should merge this and then tackle that issue.